### PR TITLE
Default configuration file

### PR DESCRIPTION
**Pull request by gdraynz**

### CODE REVIEW

File: nyuki/config.py [hunk 1]
```diff
@@ -53,16 +52,28 @@ def merge_configs(*configs):
     return new_conf
 
 
+def read_default_json(return_json=True):
+    """
+    Read the default configuration file
+    No error check, just break everything is it not available
+    """
+    with open(DEFAULT_CONF_FILE, 'r') as f:
```
Comment (Ketouem) [inline nyuki/config.py]: A `FileNotFoundError` will be thrown if `default.json` does not exist. Such case should be a deal breaker and stop the nyuki.

File: tests/nyuki_test.py [hunk 2]
```diff
@@ -102,3 +84,23 @@ def test_005b_custom_schema_ok(self):
...
+    def tearDown(self):
+        if os.path.exists('something.json'):
+            os.remove('something.json')
+        if os.path.exists('default.json'):
+            os.remove('default.json')
+
+    @ignore_loop
+    def test_001_copy_default(self):
+        with assert_raises(FileNotFoundError):
+            open('something.json')
```
Comment (Ketouem) [inline tests/nyuki_test.py]: Stage this stuff in a proper temp folder, there is some tooling inside the standard library using [tempfile](https://docs.python.org/3/library/tempfile.html)

File: tests/nyuki_test.py [hunk 3]
```diff
@@ -102,3 +84,23 @@ def test_005b_custom_schema_ok(self):
...
+class TestNyukiWithConfig(TestCase):
+
+    def tearDown(self):
+        if os.path.exists('something.json'):
+            os.remove('something.json')
+        if os.path.exists('default.json'):
+            os.remove('default.json')
+
+    @ignore_loop
+    def test_001_copy_default(self):
```
Comment (Ketouem) [inline tests/nyuki_test.py]: Add some more tests to cover the cases:
- missing default.json when loading with no config
- loading with a bad config file

File: nyuki/config.py (hunk 1 above)
Comment (gdraynz) [inline nyuki/config.py]: `FileNotFoundError` does indeed raise here and break the nyuki's `__init__`

**Pull request merged**